src/Components/Header/Header.tsx

```diff
@@ -0,0 +1,71 @@
+import classNames from 'classnames';
+import React, {
+  ChangeEvent,
+  FormEvent,
+  useRef,
+  useState,
+} from 'react';
+
+type Props = {
+  todosLength: number;
+  completedTodosLength: number;
+  addNewTodo: (value: string) => Promise<void>;
+};
+
+const Header: React.FC<Props> = ({
+  todosLength,
+  completedTodosLength,
+  addNewTodo,
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const [value, setValue] = useState('');
+  const [isSubmitValue, setSubmitValue] = useState(false);
+
+  const onChangeValue = (e: ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+  };
+
+  const onSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    setSubmitValue(true);
+    addNewTodo(value)
+      .finally(() => {
+        setValue('');
+        setSubmitValue(false);
+        inputRef.current?.focus();
+      });
+  };
```

Comment:
As was told in fe_chat. Try to use useEffect for focus. Main idea is that rerender reset your focus